packages/contracts/contracts/L1/rollup/OVM_CanonicalTransactionChain.sol

```diff
@@ -412,16 +412,6 @@ contract OVM_CanonicalTransactionChain is iOVM_CanonicalTransactionChain, Lib_Ad
             "Function can only be called by the Sequencer."
         );
 
-        require(
-            numContexts > 0,
-            "Must provide at least one batch context."
-        );
-
-        require(
-            totalElementsToAppend > 0,
-            "Must append at least one element."
-        );
```

Comment:
@tynes are there any new things the batch submitter could screw up with these checks gone?  I don't think so but worth a second set of eyes.

Reply:
Assuming no misconfiguration, the batch submitter will not submit empty batches. I don't think it would be a bad idea to prevent the batch submitter from running with this bad config.

https://github.com/ethereum-optimism/optimism/blob/349279289076dd250e188f863d0b656088e57afd/packages/batch-submitter/src/exec/run-batch-submitter.ts#L278

A check could be added here:
https://github.com/ethereum-optimism/optimism/blob/349279289076dd250e188f863d0b656088e57afd/packages/batch-submitter/src/exec/run-batch-submitter.ts#L341

Like so:

```js
if (requiredEnvVars.MIN_L1_TX_SIZE === 0) {
  logger.error('Must configure a MIN_L1_TX_SIZE greater than 0')
  process.exit(1)
}

```

Reply:
Agreed with you @tynes . Not a bad idea to add that check to `run-batch-submitter.ts` to avoid broken configuration. That said, there isn't much risk of the batch submitter 'messing up', even if it were to submit an empty batch it would just be burning gas, not triggering some super bad protocol state.

Reply:
Implemented here: https://github.com/ethereum-optimism/optimism/pull/1469